scripts/Diffraction/isis_powder/routines/common.py

```diff
@@ -0,0 +1,165 @@
+
+def set_debug(debug_on=False):
+    global g_debug
+    g_debug = debug_on
+
+
+def remove_intermediate_workspace(workspace_name):
+    _remove_ws(ws_to_remove=workspace_name)
+
+# --- Private Implementation --- #
+
+# Please note these functions can change in any way at any time.
+# For this reason please do not call them directly and instead use the Public API provided.
+
+# If this doesn't quite provide what you need please let a developer know so we can create
+# another API or update an existing one
+
+
+_read_ws_count = 0
+global g_ads_workaround
+g_ads_workaround = {"read_ws": _read_ws_count}
+
+
+def _create_blank_cal_file(calibration_runs, out_grouping_file_name, instrument, group_names):
+    input_ws = load_current_normalised_ws(calibration_runs, instrument)
+    calibration_d_spacing_ws = mantid.ConvertUnits(InputWorkspace=input_ws, Target="dSpacing")
+    mantid.CreateCalFileByNames(InstrumentWorkspace=calibration_d_spacing_ws,
+                                GroupingFileName=out_grouping_file_name, GroupNames=group_names)
+    remove_intermediate_workspace(calibration_d_spacing_ws)
+    remove_intermediate_workspace(input_ws)
+
+
+def load_monitor(run_numbers, instrument):
+    number_list = generate_run_numbers(run_numbers)
+    monitor_spectra = instrument._get_monitor_spectra(number_list[0])
+
+    if len(number_list) == 1:
+        file_name = instrument._generate_inst_file_name(run_number=number_list[0])
+        load_monitor_ws = mantid.Load(Filename=file_name, LoadLogFiles="0",
+                                      SpectrumMin=monitor_spectra, SpectrumMax=monitor_spectra)
+    else:
+        load_monitor_ws = _load_sum_file_range(run_numbers_list=number_list, instrument=instrument)
+        load_monitor_ws = mantid.ExtractSingleSpectrum(InputWorkspace=load_monitor_ws, WorkspaceIndex=monitor_spectra)
+
+    return load_monitor_ws
+
+
+def subtract_sample_empty(ws_to_correct, empty_sample_ws_string, instrument):
+    output = ws_to_correct
+    if empty_sample_ws_string is not None:
+        empty_sample = load_current_normalised_ws(run_number_string=empty_sample_ws_string, instrument=instrument)
+        corrected_ws = mantid.Minus(LHSWorkspace=ws_to_correct, RHSWorkspace=empty_sample)
+        remove_intermediate_workspace(empty_sample)
+        output = corrected_ws
+    return output
+
+
+def _load_raw_files(run_number_string, instrument):
+    run_number_list = generate_run_numbers(run_number_string=run_number_string)
+    instrument.PEARL_setup_input_directories(run_number=run_number_list[0])
+    load_raw_ws = _load_sum_file_range(run_number_list, instrument)
+    return load_raw_ws
+
+
+def _load_sum_file_range(run_numbers_list, instrument):
+    read_ws_list = []
+
+    _check_load_range(list_of_runs_to_load=run_numbers_list)
+
+    for run_number in run_numbers_list:
+        file_name = instrument._generate_inst_file_name(run_number=run_number)
+        read_ws = mantid.Load(Filename=file_name, LoadLogFiles="0")
```

Comment:
I think you can give `Load` a list of filenames to load which would give you a group workspace. `Load` also checks if more than 100 (?) files are given, so `_check_load_range` would not be needed. However, I don't know if you would be happy with a group workspace. Anyway, might be worth checking that option out.